### PR TITLE
[lldb][AIX] AIX Changes for MainLoop polling

### PR DESCRIPTION
**Pull request by DhruvSrivastavaX**
This PR is in reference to porting LLDB on AIX.

Link to discussions on llvm discourse and github:

1. https://discourse.llvm.org/t/port-lldb-to-ibm-aix/80640
2. https://github.com/llvm/llvm-project/issues/101657
The complete changes for porting are present in this draft PR:
https://github.com/llvm/llvm-project/pull/102601

Dropping changes for MainLoop polling in AIX, as `ppoll` is not supported in AIX currently. 
This change is part of the couple of minimal changes required to build a minimal `lldb` binary on AIX

Review Request: @labath @DavidSpickett

### CODE REVIEW

File: lldb/source/Host/posix/MainLoopPosix.cpp [hunk 1]
```diff
@@ -226,13 +238,13 @@ MainLoopPosix::~MainLoopPosix() {
...
   m_interrupt_pipe.Close();
-  assert(m_read_fds.size() == 0); 
+  assert(m_read_fds.size() == 0);
   assert(m_signals.size() == 0);
 }
 
 MainLoopPosix::ReadHandleUP
 MainLoopPosix::RegisterReadObject(const IOObjectSP &object_sp,
-                                 const Callback &callback, Status &error) {
+                                  const Callback &callback, Status &error) {
```
Comment (DhruvSrivastavaX) [inline lldb/source/Host/posix/MainLoopPosix.cpp:247]: Above two changes are done by clang-format

Comment (DhruvSrivastavaX) [inline lldb/source/Host/posix/MainLoopPosix.cpp:247]: Above changes are done by clang-format

**Comment by DavidSpickett** [comment]
I just pushed the clang-format changes for you, rebase to include them.

**Comment by DavidSpickett** [comment]
You can send a formatting PR then one for actual changes, where the first commit is the formatting. You'll just need to rebase the second one once the first one lands. This will save us some back and forth.

You can use one of those stacked PR manager tools but for <= 2 changes I would just do it manually.

**Review by labath — COMMENTED** [review]
Small patch, but a lot of comments:
- with the last round of MainLoop changes, the sigmask changes aren't necessary anymore -- just drop them (this also sort of means we could just use `poll` everywhere, but I kinda like it because of the increased precision.
- you need to used the timeout field -- required due the last set of changes
- let's make this conditional on `HAVE_PPOLL` instead of `!_AIX`
- let's move the code into a helper function

Putting everything together, I think the code should be something like this:

```
// If nfds_t doesn't exist on AIX, just use size_t
static int Poll(ArrayRef<struct pollfd> fds, std::optional<MainLoopPosix::TimePoint> point) {
#ifdef HAVE_PPOLL
  return ppoll(fds.data(), fds.size(),
            ToTimeSpec(loop.GetNextWakeupTime()),
            /*sigmask=*/nullptr);
#else
  using namespace std::chrono;
  int timeout = -1;
  if (point) {
    nanosecond dur = std::max(*point - steady_clock::now(), nanoseconds(0));
    timeout = ceil<milliseconds>(dur).count();
  }
  return poll(fds.data(), fds.size(), timeout);
#endif
}
```

Please also make sure all of the MainLoop unit tests pass with this change. Ideally on AIX, but if you still can't get even the unit test binary to build then at least by "simulating" it by undefining the HAVE_PPOLL macro.

**Comment by DhruvSrivastavaX** [comment]
Okay sure. Thats a great suggestion.
Updated my branch for this one.

File: lldb/source/Host/posix/MainLoopPosix.cpp [hunk 2]
```diff
@@ -159,6 +160,22 @@ MainLoopPosix::RunImpl::RunImpl(MainLoopPosix &loop) : loop(loop) {
   read_fds.reserve(loop.m_read_fds.size());
 }
 
+int MainLoopPosix::RunImpl::StartPoll(
+    std::optional<MainLoopPosix::TimePoint> point) {
+#if HAVE_PPOLL
```
Comment (DhruvSrivastavaX) [inline lldb/source/Host/posix/MainLoopPosix.cpp:165]: I noticed that HAVE_PPOLL is always defined but only its value will either be 0 or 1, so added accordingly.

**Comment by DhruvSrivastavaX** [comment]
> Putting everything together, I think the code should be something like this:

Thanks for the suggestions. I have modified the implementation.

**Review by labath — APPROVED** [review]
For me personally this would look better as thin syscall wrapper -- a static function with all arguments passed explicitly (since the arguments are modified, it would have to be MutableArrayRef, and not ArrayRef as I originally wrote), rather than a method, but this will work too.

**Comment by DhruvSrivastavaX** [comment]
> For me personally this would look better as thin syscall wrapper -- a static function with all arguments passed explicitly (since the arguments are modified, it would have to be MutableArrayRef, and not ArrayRef as I originally wrote), rather than a method, but this will work too.

Okay sure. That will increase the utility of the API. Done!

**Comment by DhruvSrivastavaX** [comment]
Request you to merge it as well, if all is good. Thanks!

**Pull request merged**